libp2p/network/url_conversion.go

```diff
@@ -0,0 +1,107 @@
+package network
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+	"strings"
+
+	"github.com/multiformats/go-multiaddr"
+)
+
+const (
+	// anyScheme is a string which fills the "scheme" degree of freedom for
+	// a URL. Used to pad URLs without schemes such that they can be parsed
+	// via stdlib URL parser.
+	anyScheme = "scheme://"
+)
+
+// Libp2pMultiaddrFromServiceUrl transforms a URL into its libp2p multiaddr equivalent.
+// (see: https://github.com/libp2p/specs/blob/master/addressing/README.md#multiaddr-basics)
+func Libp2pMultiaddrFromServiceUrl(serviceUrl string) (multiaddr.Multiaddr, error) {
+	peerUrl, err := url.Parse(anyScheme + serviceUrl)
+	if err != nil {
+		return nil, fmt.Errorf(
+			"parsing peer service URL: %s: %w",
+			serviceUrl,
+			err,
+		)
+	}
+
+	var (
```

Comment:
Thoughts on putting this at the top of the file (outside the function)?

It'll help capture the TECHDEBT and current state of things immediately.

Reply:
For sure, there's no reason for this to be reallocated each call - I just missed this. :sweat_smile:  Nice catch! :raised_hands: